plugins/design-system/components/component-example/ComponentExample.js

```diff
@@ -0,0 +1,244 @@
+import React, { Component } from "react";
+import ReactDOMServer from "react-dom/server";
+import reactElementToJSXString from "react-element-to-jsx-string";
+
+import StringUtil from "#SRC/js/utils/StringUtil";
+import ClipboardTrigger from "#SRC/js/components/ClipboardTrigger";
+import Icon from "#SRC/js/components/Icon";
+
+import CodeExample from "./CodeExample";
+import CodeExampleFooter from "./CodeExampleFooter";
+import CodeExampleHeader from "./CodeExampleHeader";
+
+import CodeExampleTabButtonList from "./CodeExampleTabButtonList";
+import CodeExampleTabs from "./CodeExampleTabs";
+import CodeExampleTabView from "./CodeExampleTabView";
+import CodeExampleTabViewList from "./CodeExampleTabViewList";
+
+import CodeExampleTabButton from "./CodeExampleTabButton";
+
+import { ReactCode, HtmlCode, Preview } from "./ComponentExampleOverrides";
+
+const DEFAULT_HEIGHT = 205;
+
+const REACT_STRING_OPTIONS = {
+  showDefaultProps: false,
+  useBooleanShorthandSyntax: false
+};
+
+const TAB_LABELS = {
+  React: "REACT",
+  Html: "HTML"
+};
+
+const METHODS_TO_BIND = [
+  "expandCollapseToggle",
+  "handleTabChange",
+  "handleCodeCopy",
+  "handleCanExpand"
+];
+
+class ComponentExample extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      isExpanded: false,
+      isCodeCopied: false,
+      canExpand: false,
+      activeTab: this.isTabbedMode() ? this.props.activeTab || ReactCode : null
+    };
+
+    this.defaultHeight = this.props.defaultHeight || DEFAULT_HEIGHT;
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  findChildByType(type) {
```

Comment:
Wouldn't it be better introducing props for `Preview`, `ReactCode` and `HtmlCode` instead of having a rather weird looking `ComponentExampleOverrides` and constantly looping though the children to find a one of them?